tests/unit/gaze/io/asc_test.py

```diff
@@ -131,64 +131,128 @@
     ],
 )
 def test_from_asc_has_shape_and_schema(kwargs, expected_frame):
-    gaze, _ = pm.gaze.from_asc(**kwargs)
+    gaze = pm.gaze.from_asc(**kwargs)
 
     assert_frame_equal(gaze.frame, expected_frame, check_column_order=False)
 
 
 @pytest.mark.parametrize(
-    ('kwargs', 'expected_metadata'),
+    ('kwargs', 'exception', 'message'),
     [
         pytest.param(
             {
                 'file': 'tests/files/eyelink_monocular_example.asc',
-                'metadata_patterns': [
-                    {'pattern': r'!V TRIAL_VAR SUBJECT_ID (?P<subject_id>-?\d+)'},
-                    r'!V TRIAL_VAR STIMULUS_COMBINATION_ID (?P<stimulus_combination_id>.+)',
-                ],
+                'patterns': 'foobar',
             },
+            ValueError,
+            "unknown pattern key 'foobar'. Supported keys are: eyelink",
+            id='unknown_pattern',
+        ),
+    ],
+)
+def test_from_asc_raises_exception(kwargs, exception, message):
+    with pytest.raises(exception) as excinfo:
+        pm.gaze.from_asc(**kwargs)
+
+    msg, = excinfo.value.args
+    assert msg == message
```

Comment:
this seems to be only one test so we do not need parametrize

Reply:
I did not touch this test :sweat_smile: (the diff is a bit messed up here...)